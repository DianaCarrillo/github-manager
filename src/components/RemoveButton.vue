<template>
  <div>
    <button id="btn-remove"  @click="removeReaction">Click to remove HOORAY</button>
  </div>
</template>

<script>
import gql from 'graphql-tag'

const addReactionIssue = gql`
  mutation AddReactionToIssue($input: AddReactionInput!) {
    addReaction(input: $input) {
      reaction {
        id
        content
      }
      subject {
        id
      }
    }
  }
`
export default {
  name: 'RemoveButton',
  props: {
    selected: {
      type: Object,
      required: true,
    },
  },

  methods: {
    addReaction: function() {
      this.$apollo.mutate({
        mutation: addReactionIssue,
        variables: {
          input: { subjectId: this.selected.id, content: 'HOORAY' },
        },
      })
    },
  },

  apollo: {
    repository: {
      query: gql`
        query FindIssueID($owner: String!, $name: String!, $number: Int!) {
          repository(owner: $owner, name: $name) {
            issue(number: $number) {
              __typename
              title
              id
              body
            }
            name
          }
        }
      `,
      variables() {
        return {
          owner: 'DianaCarrillo',
          name: 'cdmx-2018-01-FE-markdown',
          number: this.selected.number,
        }
      },
    },
  },
}
</script>

<style>
#btn-remove{
  padding: 10px;
  margin: 1%;
  color: white;
  background-color:red;
  border-color: red
}
</style>
