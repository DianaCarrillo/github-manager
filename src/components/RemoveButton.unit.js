import RemoveButton from './RemoveButton.vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import gql from 'graphql-tag'

describe('./RemoveButton.vue', () => {
  it('called Apollo mutation in addReaction method', () => {
    let localVue = createLocalVue()
    const mutate = jest.fn()
    const wrapper = shallowMount(RemoveButton, {
      propsData: {
        selected: {
          author: {
            avatarUrl: 'https://avatars0.githubusercontent.com/u/39560661?v=4',
            login: 'DianaCarrillo',
          },
          body:
            'Este Issue solamente es para recordarnos que para mañana esperamos poder: -empezar a imprimir la publicación -poder editar -poder borrar nuestra publicación.',
          id: 'MDU6SXNzdWUzNDQxODAyOTI=',
          number: 15,
          state: 'CLOSED',
          title: 'Para el sprint 2',
        },
      },
      localVue,
      mocks: {
        $apollo: {
          mutate,
        },
      },
    })
    wrapper.vm.removeReaction()
    expect(mutate).toBeCalled()
  })

  it('ReactionButton exists', () => {
    let localVue = createLocalVue()
    const wrapper = shallowMount(RemoveButton, {
      propsData: {
        selected: {
          author: {
            avatarUrl: 'https://avatars0.githubusercontent.com/u/39560661?v=4',
            login: 'DianaCarrillo',
          },
          body:
            'Este Issue solamente es para recordarnos que para mañana esperamos poder: -empezar a imprimir la publicación -poder editar -poder borrar nuestra publicación.',
          id: 'MDU6SXNzdWUzNDQxODAyOTI=',
          number: 15,
          state: 'CLOSED',
          title: 'Para el sprint 2',
        },
      },
      localVue,
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('Renders repository query', () => {
    const repo = gql`
      query FindIssueID {
        repository(owner: "DianaCarrillo", name: "cdmx-2018-01-FE-markdown") {
          issue(number: 7) {
            __typename
            title
            id
            body
          }
          name
        }
      }
    `
    const query = jest.fn(async () => {
      await repo
    })
    const wrapper = shallowMount(RemoveButton, {
      propsData: {
        selected: {
          author: {
            avatarUrl: 'https://avatars0.githubusercontent.com/u/39560661?v=4',
            login: 'DianaCarrillo',
          },
          body:
            'Este Issue solamente es para recordarnos que para mañana esperamos poder: -empezar a imprimir la publicación -poder editar -poder borrar nuestra publicación.',
          id: 'MDU6SXNzdWUzNDQxODAyOTI=',
          number: 15,
          state: 'CLOSED',
          title: 'Para el sprint 2',
        },
      },
      mocks: {
        $apollo: {
          query,
        },
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
