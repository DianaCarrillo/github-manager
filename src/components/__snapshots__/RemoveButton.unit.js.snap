// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./RemoveButton.vue Renders repository query 1`] = `<div><button id="btnRemove" type="submit">Click to remove HOORAY</button></div>`;
